Duplicati/Library/Main/Operation/Backup/BackendUploader.cs

```diff
@@ -240,6 +224,49 @@ public Task Run()
             });
         }
 
+        private Worker GetWorker(List<Worker> workers)
+        {
+            var worker = workers.FirstOrDefault(w => w.Task.IsCompleted && !w.Task.IsFaulted);
+            if (worker == null)
+            {
+                worker = new Worker(m_backendFactory());
+                workers.Add(worker);
+            }
+
+            return worker;
+        }
+
+        private async Task WaitForWorkers(List<Worker> workers, FlushRequest flush)
+        {
+            while (workers.Any())
+            {
+                Task finishedTask = await Task.WhenAny(workers.Select(w => w.Task)).ConfigureAwait(false);
+                if (finishedTask.IsFaulted)
+                {
+                    flush.TrySetCanceled();
+                    ExceptionDispatchInfo.Capture(finishedTask.Exception).Throw();
+                }
+
+                Worker finishedWorker = workers.Single(w => w.Task == finishedTask);
+                workers.Remove(finishedWorker);
+                finishedWorker.Dispose();
+            }
+        }
+
+        private void UploadFilesets(List<Worker> workers, List<FilesetUploadRequest> filesets)
+        {
+            if (filesets.Any())

```

Comment:
Is this `if` statement necessary given the `foreach` below?

Reply:
Good catch. You're right, I'll remove it.